### PR TITLE
Fix filesystem detection for clang

### PR DESCRIPTION
**Pull request by aaronmondal**
Clang defines __GNUC__ but does not use experimental/filesystem.

### CODE REVIEW

File: nvbench/json_printer.cu [hunk 1]
```diff
@@ -16,24 +16,18 @@
  *  limitations under the License.
  */
 
-#include <nvbench/json_printer.cuh>
-
 #include <nvbench/axes_metadata.cuh>
 #include <nvbench/benchmark_base.cuh>
 #include <nvbench/config.cuh>
+#include <nvbench/detail/throw.cuh>
```
Comment (allisonvacanti) [inline nvbench/json_printer.cu:22]: Let's avoid reordering the headers in this branch and just fix the clang bug

**Review by allisonvacanti — CHANGES_REQUESTED** [review]
LGTM, but let's remove the unrelated changes. We'll be doing a large style sweep soon and I'd rather not thrash too much on these includes.

File: nvbench/json_printer.cu (hunk 1 above)
Comment (aaronmondal) [reply nvbench/json_printer.cu]: Reverted.

**Review by allisonvacanti — APPROVED** [review]

**Comment by allisonvacanti** [comment]
Thanks!

**Pull request merged**